lesson_6/alexandra_nechita/6thHomework_Python.py

```diff
@@ -0,0 +1,141 @@
+from collections import defaultdict
+
+# function that prepares the dataset
+
+def prepare_dataset (raw_data):
+    description = ['Country', ('2011 ', '2012 ', '2013 ', '2014 ', '2015 ', '2016 ', '2017 ', '2018 ', '2019 ')]
+    dataset_dict = defaultdict (list)
+    
+#Varianta 1 - with no use of the description list
+    
+#     for country, coverage in raw_data:
+#         year=2011
+#         for index in coverage:
+#             dataset_dict[country].append ({'year':year, 'coverage':index})
+#             year=year+1
+#     return dataset_dict
+
+
+#Varianta2 - using description list, multiple for loops
+
+#     for country, coverage in raw_data:
+#         for index in reversed(coverage):
+#             for year in reversed(description[1]):
+#                 dataset_dict[country].append ({'year':year, 'coverage':index})
+#     return dataset_dict
+
+#Varianta3 - using description list, only 2 for loops
+
+    for country, coverage in raw_data:
+        year_index=len(description[1])-1
+        for index in reversed(coverage):
+            dataset_dict[country].append ({'year':description[1][year_index], 'coverage':index})
+            year_index=year_index-1
+    return dataset_dict
+
+# function to retrieve data for each year
+
+def get_year_data (dataset_dict, required_year):
+    year_data_dict = defaultdict (list)
+    for country, data_per_country in dataset_dict.items():
+        for data_per_year in data_per_country:
+            if int(data_per_year['year'])==required_year:
+                year_data_dict[required_year].append((country, data_per_year['coverage']))
+    return year_data_dict
+
+# function to retrieve data for each country
+
+def get_country_data (dataset_dict, required_country):
+    country_data_dict = defaultdict (list)
+    for country, data_per_country in dataset_dict.items():
+        if country==required_country:
+            for data_per_year in data_per_country:            
+                country_data_dict[required_country].append((data_per_year['year'], data_per_year['coverage']))
+    return country_data_dict
+
+# function to perform average from an iterable(of year data or country data)
+
+# Varianta 1 - where we skip non-numeric data for coverage
```

Comment:
Imi place mai mult varianta 1 in cazul `perform_average`.  O ideie ar fi sa separam partea de clean-up ( inlocuirea spatilor  si a ': ') intr-o functie separata pe care o chemam in `perform_average` si ulterior sa folosim `sum()` https://docs.python.org/3/library/functions.html#sum pentru suma.